AxonDeepSeg/integrity_test.py

```diff
@@ -17,48 +22,70 @@
+            print("* * * Integrity test passed. AxonDeepSeg is correctly installed. * * * ")
+            return 0
 
-        # input parameters
-        path = Path('folder_name') / 'file_name'
-        model_name = 'model_seg_generalist_light'
-        path_model = dir_path / 'models' / model_name
-        path_testing = dir_path.parent / 'test' / '__test_files__' / '__test_demo_files__'
-        image = Path("image.png")
+        except IOError:
 
-        with tempfile.TemporaryDirectory() as tmpdirname:
-            path_tmp = Path(tmpdirname)
-            shutil.copy(path_testing / image, path_tmp)
+            # Else, there is a problem in the installation
+            print("Integrity test failed... ")
+            return -1
 
-            # Launch the axon and myelin segmentation on test image sample provided in the installation
-            print('Computing the segmentation of axon and myelin on test image.')
-            axon_segmentation([str(path_tmp / image)], path_model)
+def main():
+    # argparse CLI arguments, default is to run the integrity test only
 
-            # Read the ground truth mask and the obtained segmentation mask
-            mask = ads.imread(path_testing / 'mask.png')
-            pred = ads.imread(path_tmp / (image.stem + str(axonmyelin_suffix)))
+    parser = argparse.ArgumentParser(description='Run AxonDeepSeg tests.')
 
-            # Generate separate axon and myelin masks of the segmentation output
-            print('Generating axon and myelin segmentation masks and saving.')
-            gt_axon = mask > 200 # Generate binary image with the axons for the ground truth (myelin=127, axon=255)
-            gt_myelin = np.logical_and(mask >= 50, mask <= 200) # Generate binary image with the myelin for the ground truth (myelin=127, axon=255)
-            pred_axon = pred > 200 # Generate binary image with the axons for the segmentation (myelin=127, axon=255)
-            pred_myelin = np.logical_and(pred >= 50, pred <= 200) # Generate binary image with the myelin for the segmentation (myelin=127, axon=255)
+    parser.add_argument('--integrity', action='store_true', help='Run the integrity test. (default)')
+    parser.add_argument('--full', action='store_true', help='Run the the full test suite.')
 
-            # Compute Dice between segmentation and ground truth, for both axon and myelin
-            dice_axon = pw_dice(pred_axon, gt_axon)
-            dice_myelin = pw_dice(pred_myelin, gt_myelin)
+    args = parser.parse_args()
 
-        # If all the commands above are executed without bugs, the installation is done correctly
-        print("* * * Integrity test passed. AxonDeepSeg is correctly installed. * * * ")
-        return 0
+    if args.full:
+        # Run the full test suite
+        sys.exit(pytest.main(["-x", Path(__file__).resolve().parent.parent / "test"]))
 
-    except IOError:
+    else:
+        # Run the integrity test
+        integrity_test()
+    return 0
```

Comment:
Rather than `return 0` for `main()`, maybe it would be useful to preserve the result of `integrity_test()` and return that instead?

Reply:
How could that output be leveraged ? (it's already all printed for the user)

Reply:
Honestly, I don't know, hehe. I just figured it would be more accurate than `return 0` even when the tests fail? 

But maybe `return 0` is meant to signify whether the tests ran successfully, rather than whether the tests passed? If so, then ignore this comment!

Reply:
Yep I think its the latter, if other errors in the software arose, they’d produce different integer values